pkgs/tools/misc/qjoypad/default.nix

```
@@ -1,19 +1,18 @@
-{ stdenv, fetchurl, pkgconfig, libX11, libXtst, qt4 }:
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, libX11, libXtst, qt5 }:
+
 stdenv.mkDerivation rec {
-  name = "qjoypad-4.1.0";
-  src = fetchurl {
-    url = "mirror://sourceforge/qjoypad/${name}.tar.gz";
-    sha256 = "1jlm7i26nfp185xrl41kz5z6fgvyj51bjpz48cg27xx64y40iamm";
+  name = "qjoypad-${version}";
+  version = "4.3.0";
+  
+  src = fetchFromGitHub {
+    owner = "panzi";
+    repo = "qjoypad";
+    rev = "v${version}";
+    sha256 = "0ps85ncqmy8jz3zzyv5f5z4564kkgcl799ri4kcz407mjyc4fvs6";
   };
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ libX11 libXtst qt4 ];
+  nativeBuildInputs = [ pkgconfig cmake ];
+  buildInputs = [ libX11 libXtst qt5.qtbase qt5.qttools qt5.qtx11extras ];
   NIX_LDFLAGS = [ "-lX11" ];
-  patchPhase = ''
-    cd src
-    substituteInPlace config --replace /bin/bash ${stdenv.shell}
-    mkdir -p $out
-    export NIX_LDFLAGS="$NIX_LDFLAGS -rpath ${libX11}/lib"
-  '';
   meta = {
```

personal preference, but also very common
```suggestion
  meta = with lib {
```